src/library/browse/foldertreemodel.h

```diff
@@ -16,13 +34,39 @@ class TreeItem;
 
 class FolderTreeModel : public TreeItemModel {
     Q_OBJECT
+
+    using FolderQueue = QQueue<std::pair<QModelIndex, QString>>;
+    using Cache = std::unordered_map<QString, bool>;
+
   public:
-    FolderTreeModel(QObject *parent = 0);
+    using TreeItemList = QList<TreeItem*>*;
+
+    FolderTreeModel(QObject* parent = 0);
     virtual ~FolderTreeModel();
     virtual bool hasChildren(const QModelIndex& parent = QModelIndex()) const;
-    bool directoryHasChildren(const QString& path) const;
+    void processFolder(const QModelIndex& parent, const QString& path) const;
+
+  private:
+    void directoryHasChildren(const QString& path);
+
+  private slots:
+    void dirModified(const QString& str);
+    void addChildren(const QModelIndex& parent, TreeItemList children);
+    void onHasSubDirectory(const QString& path);
+
+  signals:
+    void newChildren(const QModelIndex& parent, TreeItemList hildren);
+    void hasSubDirectory(const QString& path) const;
```

Comment:
@daschuer @uklotzde this has to be const as the caller function is const... unfortunately `clazy` is complaining about this. is there a macro to skip this line's checking somehow?

Reply:
util/qt.h: thisAsNonConst()